packages/web3-eth/src/index.ts

```diff
@@ -0,0 +1,371 @@
+		hydrated: boolean,
+		returnType?: ReturnType,
+	) {
+		const response = isHexString32Bytes(block)
+			? await RpcMethods.getBlockByHash(this._requestManager, block, hydrated)
+			: await RpcMethods.getBlockByNumber(this._requestManager, block, hydrated);
+
+		return convertObjectPropertiesToValidType(
+			response,
+			convertibleBlockProperties,
+			returnType ?? this._options.defaultReturnType,
+		);
+	}
+
+	// TODO Discuss the use of multiple optional parameters
+	public async getBlockTransactionCount<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		block: HexString32Bytes | BlockNumberOrTag = this._options.defaultBlock,
+		returnType?: ReturnType,
+	) {
+		const response = isHexString32Bytes(block)
+			? await RpcMethods.getBlockTransactionCountByHash(this._requestManager, block)
+			: await RpcMethods.getBlockTransactionCountByNumber(this._requestManager, block);
+
+		return convertToValidType(
+			response,
+			returnType ?? this._options.defaultReturnType,
+		) as ValidReturnTypes[ReturnType];
+	}
+
+	// TODO Discuss the use of multiple optional parameters
+	public async getBlockUncleCount<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		block: HexString32Bytes | BlockNumberOrTag = this._options.defaultBlock,
+		returnType?: ReturnType,
+	) {
+		const response = isHexString32Bytes(block)
+			? await RpcMethods.getUncleCountByBlockHash(this._requestManager, block)
+			: await RpcMethods.getUncleCountByBlockNumber(this._requestManager, block);
+
+		return convertToValidType(
+			response,
+			returnType ?? this._options.defaultReturnType,
+		) as ValidReturnTypes[ReturnType];
+	}
+
+	public async getUncle<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		block: HexString32Bytes | BlockNumberOrTag = this._options.defaultBlock,
+		uncleIndex: Uint,
+		returnType?: ReturnType,
+	) {
+		const response = isHexString32Bytes(block)
+			? await RpcMethods.getUncleByBlockHashAndIndex(this._requestManager, block, uncleIndex)
+			: await RpcMethods.getUncleByBlockNumberAndIndex(
+					this._requestManager,
+					block,
+					uncleIndex,
+			  );
+
+		return convertObjectPropertiesToValidType(
+			response,
+			convertibleBlockProperties,
+			returnType ?? this._options.defaultReturnType,
+		);
+	}
+
+	public async getTransaction<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		transactionHash: HexString32Bytes,
+		returnType?: ReturnType,
+	) {
+		const response = await RpcMethods.getTransactionByHash(
+			this._requestManager,
+			transactionHash,
+		);
+
+		return response === null
+			? response
+			: convertObjectPropertiesToValidType(
+					response,
+					// TODO
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-expect-error
+					convertibleTransactionInfoProperties,
+					returnType ?? this._options.defaultReturnType,
+			  );
+	}
+
+	// TODO Can't find in spec
+	// public async getPendingTransactions() {
+
+	// }
```

Comment:
[Docs for 1.x](https://web3js.readthedocs.io/en/v1.2.11/web3-eth.html#getpendingtransactions)